### PR TITLE
Add some more testing around shadowing, inheritance, and project boundaries.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Initially this was meant to be a test suite to catch #86662, but now it's just regression tests for it and behavior validation tests for other scenarios I thought of at the time.

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/ComInterfaceGeneratorOutputShape.cs [hunk 1]
```diff
@@ -122,7 +123,113 @@ partial interface J : I
...
+            {
+                var j = comp.GetTypeByMetadataName("J");
+                Assert.NotNull(j);
+
+                var shadowingMethod = Assert.Single(j.GetMembers("Method"));
+                VerifyShadowingMethodShape(shadowingMethod);
+
+                shadowingMethod = Assert.Single(j.GetMembers("Method2"));
+                Assert.False(shadowingMethod.IsAbstract);
+                Assert.True(shadowingMethod.IsVirtual);
```
Comment (jtschuster) [inline src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/ComInterfaceGeneratorOutputShape.cs:170]: Should we move these asserts to VerifyShadowinMethodShape?

File: src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CompileFails.cs [hunk 2]
```diff
@@ -418,5 +419,75 @@ partial interface {|#0:IFoo2|}<T>
...
+                    .Diagnostic(GeneratorDiagnostics.BaseInterfaceIsNotGenerated)
+                    .WithLocation(0)
+                    .WithArguments("J", "I"));
+
+            // The Roslyn SDK doesn't apply the compilation options from CreateCompilationOptions to AdditionalProjects-based projects.
+            test.SolutionTransforms.Add((sln, _) =>
+            {
+                var additionalProject = sln.Projects.First(proj => proj.Name == "Other");
+                return additionalProject.WithCompilationOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, allowUnsafe: true)).Solution;
+            });
```
Comment (jtschuster) [inline src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CompileFails.cs:488]: Do you think it's worth it to make a new Test subclass for tests with AdditionProjects?

**Review by jtschuster — APPROVED** [review]
LGTM, thank you!

File: src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CompileFails.cs (hunk 2 above)
Comment (jkoritzinsky) [reply src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CompileFails.cs:488]: No, I think doing it manually for the cases where we need it is okay. It's not a lot of packages and it's easy to mess with.

Also I'll probably fix this issue in the Roslyn-SDK at some point and then we can get rid of the solution transform.

**Comment by jkoritzinsky** [comment]
Failures are all known.

**Pull request merged**